### PR TITLE
azure-http-specs, add ARM CheckNameAvailability operations case

### PR DESCRIPTION
**Pull request by v-hongli1**
### Context

- fix https://github.com/Azure/cadl-ranch/issues/680
- not covering legacy cases

### CODE REVIEW

File: packages/azure-http-specs/specs/azure/resource-manager/operation-templates/mockapi.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import { json, passOnSuccess, ScenarioMockApi } from "@typespec/spec-api";
+
+export const Scenarios: Record<string, ScenarioMockApi> = {};
+
+const SUBSCRIPTION_ID_EXPECTED = "00000000-0000-0000-0000-000000000000";
+const validCheckNameAvailability = {
+  nameAvailable: true,
+  reason: "AlreadyExists",
+  message: "",
+};
```
Comment (weidongxu-microsoft) [inline packages/azure-http-specs/specs/azure/resource-manager/operation-templates/mockapi.ts:10]: This is not same as your scenario doc

**Comment by weidongxu-microsoft** [comment]
Please resolve conflict

File: packages/azure-http-specs/specs/azure/resource-manager/operation-templates/mockapi.ts (hunk 1 above)
Comment (v-hongli1) [reply packages/azure-http-specs/specs/azure/resource-manager/operation-templates/mockapi.ts]: Fixed by the [commit#d70464f](https://github.com/Azure/typespec-azure/pull/1934/commits/d70464f7251a63363cd3b47743600f95df5e1fd9)

File: packages/azure-http-specs/specs/azure/resource-manager/operation-templates/mockapi.ts [hunk 2]
```diff
@@ -29,10 +29,61 @@ const validOrder = {
     lastModifiedByType: "User",
   },
 };
+const validCheckNameAvailability = {
```
Comment (weidongxu-microsoft) [inline packages/azure-http-specs/specs/azure/resource-manager/operation-templates/mockapi.ts:32]: nit, change the var name, e.g. checkNameAvailabilityResponse

Comment (v-hongli1) [reply packages/azure-http-specs/specs/azure/resource-manager/operation-templates/mockapi.ts]: Fixed by the [commit#a2103e4](https://github.com/Azure/typespec-azure/pull/1934/commits/a2103e423b95bc7eb27c0fe6a77b3cb37ee6776c)

**Review by weidongxu-microsoft — APPROVED** [review]

**Review by msyyc — APPROVED** [review]

**Pull request merged**